src/test/java/com/cloudant/tests/ViewsTest.java

```diff
@@ -105,12 +108,146 @@ public void byStartAndEndKey() {
+    }
+
+    @Test
+    public void byStartAndEndTwoKeyArray() {
+        multiValueKeyInit();
+
+        //Test key array with 2 (startkey=["uuid",1]&endkey=["uuid",2000])
+        View query = db.view("keyarray_example/creator_created")
+                .startKey("uuid", 1)
+                .endKey("uuid", 2000)
+                .includeDocs(true);
+
+
+        List<Object> result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+
+        Page page = null;
+        try {
+            page = query.queryPage(30, null, Object.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        assertThat(page.getResultList().size(), is(3));
+
+
+        query = db.view("keyarray_example/creator_created")
+                .startKey(new Object[]{"uuid", 1})
+                .endKey(new Object[]{"uuid", 2000})
+                .includeDocs(true);
+
+
+        result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+
+        //Key array with only integers
+        query = db.view("keyarray_example/created_total")
+                .startKey(1, 12.00)
+                .endKey(2000, 15.00)
+                .includeDocs(true);
+
+
+        result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+
+        page = null;
+        try {
+            page = query.queryPage(30, null, Object.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        assertThat(page.getResultList().size(), is(3));
+    }
+
+    @Test
+    public void byStartAndEndThreeKeyArray() {
+        multiValueKeyInit();
+
+        //Test key array with 3 (startkey=[false,"uuid",1]&endkey=[false,"uuid",2000])
+        View query = db.view("keyarray_example/creator_created_boolean")
+                .startKey(false, "uuid", 1)
+                .endKey(false, "uuid", 2000)
+                .includeDocs(true);
+
+        List<Object> result = query.query(Object.class);
+
+        assertThat(result.size(), is(1));
+
+        //Test key array with 3 (startkey=[false,"uuid",1]&endkey=["{}","uuid",2000])
+        query = db.view("keyarray_example/creator_created_boolean")
+                .startKey(false, "uuid", 1)
+                .endKey("{}", "uuid", 2000)
+                .includeDocs(true);
+
+        result = query.query(Object.class);
+
+        assertThat(result.size(), is(3));
+
+        Page page = null;
+        try {
+            page = query.queryPage(30, null, Object.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        assertThat(page.getResultList().size(), is(3));
+
+
+        query = db.view("keyarray_example/creator_created_boolean")
+                .startKey(new Object[]{true, "uuid", 1})
+                .endKey(new Object[]{true, "uuid", 2000})
+                .includeDocs(true);
+
+        result = query.query(Object.class);
+
+        assertThat(result.size(), is(2));
```

Comment:
Yes, the tearDown method is deleting the whole test database after all tests are completed.